scrabble.js

```diff
@@ -1,8 +1,458 @@
+// Scrabble Javascript Project
+
+
+// Helpful description of exporting modules: https://www.sitepoint.com/understanding-module-exports-exports-node-js/
+var TileBag = require("./tileBag.js");
+var Board = require("./board.js");
+// var Player = require("./player.js");
+
+var prompt = require('prompt');
+//start the prompt
+prompt.start();
+
+
+// ------------------------- WAVE 1 ------------------------- //
+// Scrabble constructor function (acting as a class)
 var Scrabble = function() {};
 
-// YOUR CODE HERE
-Scrabble.prototype.helloWorld = function() {
-  return 'hello world!';
+    // Test function
+    Scrabble.prototype.helloWorld = function() {
+      return 'hello world!';
+    };
+
+    // Function to score individual letters
+    Scrabble.prototype.letterScore = function(letter) {
+      var letterValues = {AEIOULNRST: 1, DG: 2, BCMP: 3, FHVWY: 4, K: 5, JX: 8, QZ: 10};
+      var letterSets = Object.keys(letterValues);
+
+      // Loop through the set of letters for each point value
+      for (var i in letterSets) {
+        // if the set with a given point value includes the letter
+        if (letterSets[i].includes(letter.toUpperCase())) {   // Tried a regex version but not sure how to put a variabel in the a-z part: (str.match(/[a-z]/i)) {  // THe i makes the reg exp case insensitive
+          // Return the point value associated with that set
+          return letterValues[letterSets[i]];
+        }
+      }
+    };
+
+    // score(word): returns the total score value for the given word
+    Scrabble.prototype.score = function(word, tiles = []) {
+      if (tiles === []) {
+        tiles = word.split('');
+      }
+      var score = 0;
+      for (var i = 0, len = word.length; i < len; i++) {
+        score = score + this.letterScore(word[i]);
+      }
+
+      // 7-letter word bonus
+      if (len == 8) {score = score + 50;}
+      return score;
```

Comment:
Where did len come from?  It looks like a global variable initialized in the for loop.  Better that's not used.  

Also why are you comparing it with 8?